devicetypes/smartenit/iot8-z.src/iot8-z.groovy

```diff
@@ -0,0 +1,223 @@
+		capability "Contact Sensor"
+		capability "Configuration"
+		capability "Refresh"
+		capability "Health Check"
+		capability "Sensor"
+		capability "Switch"
+		capability "monthpublic25501.analogSensor"
+
+		command "childOn", ["string"]
+		command "childOff", ["string"]
+
+		fingerprint manufacturer: "Smartenit, Inc", model: "IOT8-Z", deviceJoinName: "Smartenit Switch Sensor ", profileId: "0104", inClusters: "0000, 0003, 0006, 000C, 000F", outClusters: "0019" 
+	}
+}
+
+private getANALOG_INPUT_CLUSTER() { 0x000C }
+private getBINARY_INPUT_CLUETER() { 0x000F }
+private getPRESENT_VALUE_ATTRIBUTE() { 0x0055 }
+private getONOFF_ATTRIBUTE() { 0x0000 }
+
+def installed() {
+	log.debug "Installed"
+	createChildDevices()
+}
+
+def updated() {
+	log.debug "Updated"
+	refresh()
+}
+
+// parse events into attributes
+def parse(String description) {	
+	Map eventMap = zigbee.getEvent(description)
+	Map eventDescMap = zigbee.parseDescriptionAsMap(description)
+
+	if (eventMap) {
+		if (eventDescMap?.sourceEndpoint == "01" || eventDescMap?.endpoint == "01") {
+			if (eventDescMap?.clusterId == "000C") {
+				return createEvent(name: "inputValue", value: eventDescMap?.value)
+			} else {
+				sendEvent(eventMap)
+			}
+		} else {
+			def childDevice = childDevices.find {
+				it.deviceNetworkId == "$device.deviceNetworkId:${eventDescMap.sourceEndpoint}" || it.deviceNetworkId == "$device.deviceNetworkId:${eventDescMap.endpoint}"
+			}
+			if (childDevice) {
+				childDevice.sendEvent(eventMap)
+			} else {
+				log.debug "Child device: $device.deviceNetworkId:${eventDescMap.sourceEndpoint} was not found"
+			}
+		}
+	} else if (eventDescMap) {
+		if (eventDescMap?.sourceEndpoint == "01" || eventDescMap?.endpoint == "01") {
+			if (eventDescMap?.clusterId == "000F" || eventDescMap?.cluster == "000F") {
+				if (eventDescMap?.value == "00") {
+					return createEvent(name: "contact", value: "open")
+				} else if (eventDescMap?.value == "01") {
+					return createEvent(name: "contact", value: "closed")
+				}
+			} else if (eventDescMap?.clusterId == "000C" || eventDescMap?.cluster == "000C") {
+				long convertedValue = Long.parseLong(eventDescMap?.value, 16)
+				Float percentage = Float.intBitsToFloat(convertedValue.intValue())
+				percentage = (percentage / 1.60) * 100.0
+				def ceilingVal = Math.ceil(percentage)
+				if (ceilingVal > 100.0) {
+					ceilingVal = 100.0
+				}
+                
+				int intValue = (int) ceilingVal
+				return createEvent(name: "inputValue", value: intValue)
+			}
+		} else {
+			def childDevice = childDevices.find {
+				it.deviceNetworkId == "$device.deviceNetworkId:${eventDescMap.sourceEndpoint}" || it.deviceNetworkId == "$device.deviceNetworkId:${eventDescMap.endpoint}"
+			}
+			if (childDevice) {
+				if (eventDescMap?.clusterId == "000F" || eventDescMap?.cluster == "000F") {
```

Comment:
Each of `eventDescMap.clusterId` and `eventDescMap.cluster` have specific type, first is integer, while other is string, so one of the conditions will always be false.  
Furthermore, you can use integer value to compare it to constants you've defined above.

Reply:
I was a bit puzzled by this feedback. I believe that the `eventDescMap.clusterId` type in this case is a string based on the function call I'm using, which is `zigbee.parseDescriptionAsMap`. I took a look at other examples and found that some implementations use `zigbee.parse` which result in an integer type. 🙂

Reply:
Sorry for misleading you, `clusterInt` was the one I was looking for.
Example of usage: https://github.com/SmartThingsCommunity/SmartThingsPublic/blob/a7c980340992ca37508d42a70f354b5fd7d30048/devicetypes/smartthings/zigbee-power-meter.src/zigbee-power-meter.groovy#L58

Reply:
Ah ok. I understand now. 🙂 Thank you.